js/parts/DataLabels.js

```diff
@@ -886,13 +886,19 @@ if (seriesTypes.pie) {
                 };
                 dataLabel._pos = {
                     x: (x +
-                        options.x +
+                        ( // (#12985)
+                        ((point.options || {}).dataLabels || {}).x ||
```

Comment:
`(point.options || {}).dataLabels || {})` - is used twice in code, so it could be done once and stored as some variable